Source/WebKit/UIProcess/RemoteLayerTree/RemoteScrollingCoordinatorProxy.cpp

```diff
@@ -67,6 +67,10 @@ ScrollingNodeID RemoteScrollingCoordinatorProxy::rootScrollingNodeID() const
 const RemoteLayerTreeHost* RemoteScrollingCoordinatorProxy::layerTreeHost() const
 {
     DrawingAreaProxy* drawingArea = m_webPageProxy.drawingArea();
+    
+    if (!drawingArea)
+        return nullptr;
+    
     if (!is<RemoteLayerTreeDrawingAreaProxy>(drawingArea)) {
```

Comment:
I believe Matt hit the nail on the head that the compiler optimized this away because we were unconditionally dereferencing the result in ScrollingTreeScrollingNodeDelegateIOS::findActingScrollParent(). Adding the null check for the host there should be enough to have the compiler not optimize away the null check in RemoteScrollingCoordinatorProxy::layerTreeHost().